src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -702,5 +710,61 @@ protected virtual void ValidateState(OpenIdConnectProtocolValidationContext vali
                 throw LogHelper.LogException<OpenIdConnectProtocolInvalidStateException>(LogMessages.IDX10331, validationContext.State, validationContext.ProtocolMessage.State);
             }
         }
+
+        private void InitializeHashAlgorithmMap()
+        {
+            ISet<string> securityAlgorithms256 = new HashSet<string>();
+            securityAlgorithms256.Add(SecurityAlgorithms.EcdsaSha256);
+            securityAlgorithms256.Add(SecurityAlgorithms.HmacSha256);
+            securityAlgorithms256.Add(SecurityAlgorithms.RsaSha256);
+            securityAlgorithms256.Add(SecurityAlgorithms.RsaSsaPssSha256);
+            securityAlgorithms256.Add(SecurityAlgorithms.RsaSha256Signature);
+            securityAlgorithms256.Add(SecurityAlgorithms.EcdsaSha256Signature);
+            _hashAlgorithmMap.Add(securityAlgorithms256, SHA256.Create());
+
+            ISet<string> securityAlgorithms384 = new HashSet<string>();
+            securityAlgorithms384.Add(SecurityAlgorithms.EcdsaSha384);
+            securityAlgorithms384.Add(SecurityAlgorithms.HmacSha384);
+            securityAlgorithms384.Add(SecurityAlgorithms.RsaSha384);
+            securityAlgorithms384.Add(SecurityAlgorithms.RsaSsaPssSha384);
+            securityAlgorithms384.Add(SecurityAlgorithms.RsaSha384Signature);
+            securityAlgorithms384.Add(SecurityAlgorithms.EcdsaSha384Signature);
+            _hashAlgorithmMap.Add(securityAlgorithms384, SHA384.Create());
+
+            ISet<string> securityAlgorithms512 = new HashSet<string>();
+            securityAlgorithms512.Add(SecurityAlgorithms.RsaSha512);
+            securityAlgorithms512.Add(SecurityAlgorithms.EcdsaSha512);
+            securityAlgorithms512.Add(SecurityAlgorithms.HmacSha512);
+            securityAlgorithms512.Add(SecurityAlgorithms.RsaSsaPssSha512);
+            securityAlgorithms512.Add(SecurityAlgorithms.RsaSha512Signature);
+            securityAlgorithms512.Add(SecurityAlgorithms.EcdsaSha512Signature);
+            _hashAlgorithmMap.Add(securityAlgorithms512, SHA512.Create());
+        }
+
+        /// <summary>
+        /// This method is used in derived class which customizes the specific HashAlgorithmMap.
+        /// </summary>
+        /// <param name="hashAlgorithmMap">Customized HashAlgorithmMap.</param>
+        /// <param name="isAppend">Appends the customized HashAlgorithmMap to the default HashAlgorithmMap or not.</param>
+        protected void CustomizeHashAlgorithmMap(IDictionary<ISet<string>, HashAlgorithm> hashAlgorithmMap, bool isAppend = false)
```

Comment:
The intention here is to allow users to modify the default hashAlgorithm map. Why do we need a method for that? Users can directly modify the map as `HashAlgorithm` property is public. Not sure I understand why this is meant for derived classes.